numpy/core/src/multiarray/array_method.c

```diff
@@ -756,6 +756,131 @@ boundarraymethod__simple_strided_call(
 }
 
 
+/*
+ * TODO: Currently still based on the old ufunc system and not ArrayMethod!
+ *       This requires fixing the ufunc code first.
+ *
+ * Support for masked inner-strided loops.  Masked inner-strided loops are
+ * only used in the ufunc machinery.  So this special cases them.
+ * In the future it probably makes sense to create an::
+ *
+ *     Arraymethod->get_masked_strided_loop()
+ *
+ * Function which this can wrap instead.
+ */
+typedef struct {
+    NpyAuxData base;
+    PyUFuncGenericFunction unmasked_stridedloop;
+    void *innerloopdata;
+    int nargs;
+    char *dataptrs[];
+} _masked_stridedloop_data;
+
+
+static void
+_masked_stridedloop_data_free(NpyAuxData *auxdata)
+{
+    _masked_stridedloop_data *data = (_masked_stridedloop_data *)auxdata;
+    PyMem_Free(data);
+}
```

Comment:
This is a bit odd. Why use `auxdata` to point to `_masked_stridedloop_data`, why not  just free `auxdata`?

Reply:
Hehe, could change. It is because the followup inserts another `NPY_AUXDATA_FREE(data->original_auxdata)` (or something like that).